include/exadg/compressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -94,15 +94,44 @@ PostProcessor<dim, Number>::do_postprocessing(VectorType const &     solution,
                                               double const           time,
                                               types::time_step const time_step_number)
 {
-  reinit_additional_fields(solution);
+  invalidate_derived_fields();
 
   /*
    *  write output
    */
   if(output_generator.time_control.needs_evaluation(time, time_step_number))
   {
+    std::vector<dealii::SmartPointer<SolutionField<dim, Number>>> additional_fields_vtu;
+
+    if(pp_data.output_data.write_pressure)
+    {
+      pressure.evaluate(solution);
+      additional_fields_vtu.push_back(&pressure);
+    }
+    if(pp_data.output_data.write_velocity)
+    {
+      velocity.evaluate(solution);
+      additional_fields_vtu.push_back(&velocity);
+
+      if(pp_data.output_data.write_vorticity)
+      {
+        vorticity.evaluate(velocity.get());
```

Comment:
I don't understand this. I think we we have such a dependency, which is why I placed these two if-statements inside the if-statement for the velocity vector.

We can certainly debate which logic is the better one, i.e. should the user be forced to activate `write_velocity` in order to be able to do `write_vorticity`, or should the user just set `write_vorticity` and the code has to care about how to realize this. If you want, I can again write these if-statements outside the velocity, and then write `vorticity.evaluate(velocity.evaluate_get(solution));`, which will not perform unnecessary computations in case the velocity has already been evaluated before. Should I change this again?

Reply:
Yes, from my perspective this should be changed. This also means that the previous velocity.evaluate(solution) can be removed again. 
I think similar code also appears in the incompressible NSE post-processor.

Reply:
I changed back to the previous logic here: https://github.com/exadg/exadg/pull/274/commits/be5a1044ec6717e6e986b4bed0e1074ee36243fe